docs/2022-07-18-JSON-RPC-Server-API.md

```diff
@@ -279,6 +279,14 @@ If the implication cannot be shown to hold, `satisfiable` is false.
 
 In some cases, a unifier `condition` for the implication can still be provided, although the implication cannot be shown to be true.
 
+The endpoint implements the following specification:
+- `antecedent` is `\\bottom` (implication holds trivially) => `satisfiable = True` and `condition = \\bottom`
+- `consequent` is `\\bottom` and `antecedent` is not `\\bottom` => `satisfiable = False` and `condition = \\bottom`
+- implication holds (not trivial) => `satisfiable = True` and `condition /= \\bottom`
+- implication doesn't hold and the two terms do not unify,
+indicating that the program configuration can be rewritten further => `satisfiable = False` and `condition` is omitted
+- implication doesn't hold and the two terms unify, indicating that the configuration is stuck => `satisfiable = False` and `condition /= \\bottom`
+
```

Comment:
The main property we should have hold is that `SUBST(CONSEQUENT) #And PREDICATE` (where `SUBST` is the returned substitution, and `PREDICATE` is the returned predicate) should be equivalent to `ANTECEDENT` (in the matching logic sense of them representing the same sets).

Also, we should consider this to be `valid`, not `satisfiable`, I believe? Satisfiable usually means there is _some_ substitution that makes the above property true, but we want it to be true for _any_ substitution.

If it's actually just satisfiable, then we only have partial subsumption.

Ping @ana-pantilie .

Reply:
@ehildenb We can keep this in mind when designing the check-implication endpoint for the new backend. Right now this implements a workaround to make it easier for users to discern between the algorithm's output. The algorithm wasn't designed with returning the unification condition in mind.

@jberthold: `satisfiable` might not be 100% correct, we can probably change that, it's just the name of the field
@goodlyrottenapple: `check-implication` is a bad name for this procedure (`check-subsumption` would be better)

Reply:
Makes sense.